src/map/skill.c

```diff
@@ -2191,6 +2195,9 @@ static int skill_onskillusage(struct map_session_data *sd, struct block_list *bl
 	if( sd == NULL || !skill_id )
 		return 0;
 
+	// Preserve auto-cast tpye if bAutoSpellOnSkill was triggered by a skill which was cast by Abracadabra, Improvised Song or an item.
```

Comment:
typo in some comments ``tpye`` -> ``type``

Reply:
Done.